lib/newProfile.mli

```diff
@@ -50,3 +50,15 @@ val pause : unit -> accu option
 val resume : accu -> unit
 (** Profiling must not be active.
     Activates profiling with the given state. *)
+
+type sums
+(** Timings for sub-events *)
+
+val with_profiling : (unit -> 'a) -> MiniJson.t list * sums * 'a
+(** Runs the given function with profiling active and returns the
+    produced events and sum times of subevents. Profiling must not be active.
+    The function should not raise exceptions. *)
```

Comment:
Wouldn't it be better to just wrap the call with a handler that cleans up the profiling before reraising the exception?  E.g., so that if there's an uncaught not_found anomaly, this doesn't break future profiling

Reply:
Yes
The original use case with `par:` has the called function catch all exceptions but we should be more robust.